src/main/java/com/epam/ta/reportportal/core/hierarchy/AbstractFinishHierarchyHandler.java

```diff
@@ -122,9 +124,14 @@ protected void updateDescendantWithoutChildren(Long itemId, StatusEnum status, L
 		}
 	}
 
-	protected void updateDescendantWithChildren(Long itemId, LocalDateTime endTime) {
+	protected void updateDescendantWithChildren(Long itemId, LocalDateTime endTime, List<ItemAttributePojo> itemAttributes) {
 		boolean isFailed = testItemRepository.hasDescendantsWithStatusNotEqual(itemId, PASSED);
 		testItemRepository.updateStatusAndEndTimeById(itemId, isFailed ? FAILED : PASSED, endTime);
+		itemAttributes.add(new ItemAttributePojo(itemId, ATTRIBUTE_KEY_STATUS, ATTRIBUTE_VALUE_INTERRUPTED, false));
+		if (itemAttributes.size() >= INSERT_ATTRIBUTES_BATCH_SIZE) {
```

Comment:
Please, make it somehow in one method to clearly understand what it is created for.

Reply:
updated